mlflow/sagemaker/__init__.py

```diff
@@ -266,7 +291,53 @@ def deploy(app_name, model_path, execution_role_arn=None, bucket=None, run_id=No
             archive=archive,
             instance_type=instance_type,
             instance_count=instance_count,
-            vpc_config=vpc_config)
+            vpc_config=vpc_config,
+            flavor=deployment_flavor)
+
+
+def _get_or_validate_deployment_flavor(model_config, flavor=None):
+    """
+    Validates and or obtains the name of a flavor that is compatible with the model
+    and will be used by the container for deployment. If the specified flavor is
+    unsupported, the specified flavor is not contained in the model, or the model does not
+    contain any supported flavors, an exception will be thrown.
+
+    :param model_config: An MLflow Model object
+    :param flavor: The name of the flavor to validate. If not `None`, this flavor will be
+                   validated; we will determine if it is supported and contained in the model.
+                   If `None`, a flavor will be selected from the set of flavors contained in
+                   the model and returned, if possible.
+
+    :return: The name of the flavor to be used for deployment.
+    """
+    default_flavor = get_serving_flavor(model_config)
+    if default_flavor is None:
```

Comment:
I think this can be simplified:
```
if flavor is not None and flavor not in SUPPORTED_DEPLOYMENT_FLAVORS:
      raise(unsupported flavor {flavor})
flavor = flavor or get_serving_flavor(model_config)
if flavor is None:
  raise(no deployable flavor)
```

Reply:
Is this a simplification for the whole block? (i.e. both `if/elif` blocks)

Reply:
yes, the whole block can be reduced to these 5 lines imho

Reply:
I broke the `_get_or_validate...` method into two methods: `_get_preferred_deployment_flavor` and `_validate_deployment_flavor` - they are now called separately; the end result is more readable and covers the case where the specified flavor may not be contained in the model. Let me know what you think!